### PR TITLE
fix(Scripts/Spell) Added random whispers of the Death Knight Newbie Village Lich King

### PR DESCRIPTION
**Pull request by fangshun2004**
Fighting  creatures 28576 during a mission in the Death Knight Newbie Village triggers a random Whisper of the Lich King, which is generated by Aura 58207 triggering random spells 58208-582223，The interval between random speeches is controlled by Aura 58207
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [x] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/9679

## SOURCE:

The changes have been validated through:
- [x] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [x] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
-  ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [x] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1. Create a Death Knight
2. .go c 128979
3. Causes Creature 28576 to enter battle Receives a Whisper from the Lich King (11 second intervals, controlled by Aura 58207)

## Known Issues and TODO List:

- [x]Whispering may only be triggered when the player is attacking a mob while performing certain quests
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp [hunk 1]
```diff
@@ -1234,6 +1234,77 @@ class spell_death_knight_initiate_visual : public SpellScript
...
+    SPELL_LICHKINGDK011 = 58218,
+    SPELL_LICHKINGDK012 = 58219,
+    SPELL_LICHKINGDK013 = 58220,
+    SPELL_LICHKINGDK014 = 58221,
+    SPELL_LICHKINGDK015 = 58222,
+    SPELL_LICHKINGDK016 = 58223
+};
+
+//spell 58207 rand Whisper
+class spell_Lich_King_vo_blocker : public AuraScript
```
Comment (Kitzunu) [inline src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp:1259]: I'd say keep the script name all lowercase

File: src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp [hunk 2]
```diff
@@ -1234,6 +1234,77 @@ class spell_death_knight_initiate_visual : public SpellScript
...
+class spell_lich_king_vo_blocker : public AuraScript
+{
+    PrepareAuraScript(spell_lich_king_vo_blocker);
+
+    void HandleEffectApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        if (Player* target = GetTarget()->ToPlayer())
+        {
+            //spell 58208-58223
+            GetCaster()->CastSpell(target, urand(SPELL_LICHKINGDK001, SPELL_LICHKINGDK016), true);
```
Comment (Kitzunu) [inline src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp:1268]: You need to validate spells used in spellscripts

File: src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp [hunk 3]
```diff
@@ -1234,6 +1234,77 @@ class spell_death_knight_initiate_visual : public SpellScript
...
+    bool Validate(SpellInfo const* spellInfo) override
+    {
+        return sObjectMgr->GetBroadcastText(uint32(spellInfo->GetEffect(EFFECT_0).CalcValue())) &&
+            sSoundEntriesStore.LookupEntry(uint32(spellInfo->GetEffect(EFFECT_1).CalcValue()));
+    }
+
+    void HandleScript(SpellEffIndex /*effIndex*/)
+    {
+        if (Player* player = GetHitPlayer())
+            GetCaster()->Unit::Whisper(uint32(GetEffectValue()), player, false);
```
Comment (Kitzunu) [inline src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp:1292]:
```suggestion
            GetCaster()->Whisper(uint32(GetEffectValue()), player, false);
```

**Comment by Voxstrasza** [comment]
hi guys long time no see

this PR would close this issue i opened: #9679

**Comment by Helias** [comment]
> hi guys long time no see
> 
> this PR would close this issue i opened: #9679

thanks, I edited the PR with the reference of this issue

**Comment by sudlud** [comment]
Why close? :)

**Comment by fangshun2004** [comment]
No. I'm just cleaning up my branch

**Comment by fangshun2004** [comment]
How is the reopen button greyed out？？

**Comment by heyitsbench** [comment]
> How is the reopen button greyed out?

Because the branch was deleted.

**Review by Nyeriah — APPROVED** [review]

**Review by sudlud — APPROVED** [review]

**Pull request merged**